health/micro-ui/web/packages/modules/campaign-manager/src/pages/employee/NewAppConfiguration/components/SelectionCard.js

```
@@ -61,34 +61,58 @@ const SelectionCard = ({ field, t, props }) => {
+      moduleName: schemaParts[0]?.trim(),
+      masterName: schemaParts[1]?.trim(),
+      isValidSchema: true,
+    };
+  }, [selectionField?.isMdms, selectionField?.schemaCode]);
 
-  const { isLoading, data } = Digit?.Hooks.useCustomMDMS(
-    Digit?.ULBService?.getStateId(),
-    selectionField?.schemaCode?.split(".")[0],
-    [{ name: selectionField?.schemaCode?.split(".")[1] }],
+  // Fetch MDMS data if field has valid MDMS configuration
+  const { isLoading, data: mdmsData } = Digit?.Hooks.useCustomMDMS(
```

Biome (2.1.2)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)